### PR TITLE
net/udp: cleaned up header and documentation

### PR DESCRIPTION
**Pull request by haukepetersen**
added doxygen and removed two redundant function definitions.

### CODE REVIEW

File: sys/net/transport_layer/udp/udp.h [hunk 1]
```diff
@@ -1,16 +1,18 @@
 /**
- * UDP header
- *
- * Copyright (C) 2013  INRIA.
+ * Copyright (C) 2015  INRIA.
```
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: 2013 - 2015

File: sys/net/transport_layer/udp/udp.h [hunk 2]
```diff
@@ -1,16 +1,18 @@
 /**
```
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: This is usually not part of the doxygen (i.e. please remove one asterisk).

File: sys/net/transport_layer/udp/udp.h [hunk 3]
```diff
@@ -25,22 +27,74 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   Define the stack size used for the UDP thread
```
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: no need for "Define the" imho. (repeats below)

File: sys/net/transport_layer/udp/udp.h [hunk 4]
```diff
@@ -25,22 +27,74 @@
...
+ * @param[in] tolen         size of the destination address struct
+ *
+ * @return                  the number of bytes that were sent
+ */
 int32_t udp_sendto(int s, const void *buf, uint32_t len, int flags, sockaddr6_t *to, socklen_t tolen);
 
+/**
+ * @brief   Checks if the given socket is a valid UDP socket
+ *
+ * TODO: generalize to work also for non IPv6 configurations
```
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: You might want to turn this into a `@note`.

File: sys/net/transport_layer/udp/udp.h [hunk 5]
```diff
@@ -25,22 +27,74 @@
...
+ * @param[in] flags         UDP socket flags
+ * @param[in] to            the destination address
+ * @param[in] tolen         size of the destination address struct
+ *
+ * @return                  the number of bytes that were sent
+ */
 int32_t udp_sendto(int s, const void *buf, uint32_t len, int flags, sockaddr6_t *to, socklen_t tolen);
 
+/**
+ * @brief   Checks if the given socket is a valid UDP socket
```
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: s/Checks/Check/

File: sys/net/transport_layer/udp/udp.h (hunk 1 above)
Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: Also, you are not working for INRIA AFAIK, so I think you should put your name/affiliation there for 2015 and just leave that line alone. Did you look into the history? Only Oliver Gesch is mentioned as an author, and he never worked for INRIA either...

Comment (haukepetersen) [inline sys/net/transport_layer/udp/udp.h]: nope. Was at the airport and just took it as it was...

**Comment by haukepetersen** [comment]
addressed comments by @LudwigOrtmann

**Comment by cgundogan** [comment]
looks sensible to me

File: sys/net/transport_layer/udp/udp.h (hunk 2 above)
Comment (authmillenon) [inline sys/net/transport_layer/udp/udp.h]: Is this comment deprecated?

Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: which comment?

Comment (LudwigOrtmann) [inline sys/net/transport_layer/udp/udp.h]: *my* comment magically moved to the wrong section, yes.

**Comment by LudwigOrtmann** [comment]
Please squash.

File: sys/net/transport_layer/udp/udp.h (hunk 2 above)
Comment (authmillenon) [inline sys/net/transport_layer/udp/udp.h]: This was what I meant ;)

**Comment by haukepetersen** [comment]
squashed and rebased.

**Comment by LudwigOrtmann** [comment]
Everybody's happy => go

**Pull request merged**